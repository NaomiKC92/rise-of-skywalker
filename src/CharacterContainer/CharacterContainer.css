.character-container {
  margin: 0;
  padding: 0;
}

.grid-3 {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-gap: 1rem;
}

.fade {
  position: relative;
  width: 100%;
  min-height: 20vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}
 
.crawl-container {
  background-image: url('../images/crawl-sky.jpg');
  height: 200px;
  width: 100%;
  display: flex;
  justify-content: center;
  perspective: 200px;
  overflow: hidden;
  text-align: justify;
}


.crawling-text {
  position: relative;
  top: -700px;
  transform-origin: 50% 100%;
  font-size: 90px;
  color: yellow;
  animation: crawl 90s linear;
}

@keyframes crawl {
  0% {
    top: 250px;
    transform: rotateX(20deg) translateZ(0);
  }
  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}
