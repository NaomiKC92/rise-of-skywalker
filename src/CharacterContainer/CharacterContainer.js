import React, { Fragment } from 'react';
import CharacterCard from '../CharacterCard/CharacterCard';
import loadingImage from '../images/loadingBB8.gif';
import './CharacterContainer.css';

const CharacterContainer = ({
  characters,
  crawl,
  setupFavoriteCharacters,
  favoriteCharactersMap
}) => {
  if (characters.length === 0) {
    return (
      <React.Fragment>
        <h2>RISE OF SKYWALKER is working hard for you!</h2>
        <h2>Enjoy our BB8 acting cute while the page is loading...</h2>
        <img src={loadingImage} alt='Loading Image of BB8' />
      </React.Fragment>
    );
  }

  const allCharacters = characters.map((character, i) => {
    return (
      <CharacterCard
        character={character}
        setupFavoriteCharacters={setupFavoriteCharacters}
        favoriteCharactersMap={favoriteCharactersMap}
        key={i}
      />
    );
  });

  return (
    <div>
      <p>{crawl}</p>
      <div className='grid-3'>{allCharacters}</div>
    </div>
  );
};

export default CharacterContainer;
