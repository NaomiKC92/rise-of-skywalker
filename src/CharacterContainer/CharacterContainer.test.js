import React from 'react';
import { shallow } from 'enzyme';
import CharacterContainer from './CharacterContainer';

describe('CHARACTER CONTAINER', () => {
  let wrapper;

  const mockCharacters = [];
  const mockCrawl = '';
  const mockSetupFavoriteCharacters = jest.fn();
  const mockFavoriteCharactersMap = [];

  beforeEach(() => {
    wrapper = shallow(
      <CharacterContainer
        characters={mockCharacters}
        crawl={mockCrawl}
        setupFavoriteCharacters={mockSetupFavoriteCharacters}
        favoriteCharactersMap={mockFavoriteCharactersMap}
      />
    );
  });

  it('should match the CharacterContainer Component Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
