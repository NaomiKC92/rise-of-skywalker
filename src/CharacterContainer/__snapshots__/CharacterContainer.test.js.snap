// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CHARACTER CONTAINER should match the CharacterContainer Component Snapshot 1`] = `
<div
  className="loading-page"
>
  <h2>
    RISE OF SKYWALKER is working hard for you!
  </h2>
  <h2>
    Enjoy BB8 acting cute while the page is loading...
  </h2>
  <img
    alt="Loading Image of BB8"
    src="loadingBB8.gif"
  />
</div>
`;
